gwt-localizable-processor/src/main/java/org/gwtproject/i18n/processor/AptContext.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright © 2018 The GWT Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gwtproject.i18n.processor;
+
+import static java.util.Objects.nonNull;
+
+import com.squareup.javapoet.TypeName;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.type.ArrayType;
+import javax.lang.model.type.PrimitiveType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+import org.gwtproject.i18n.client.LocalizableResource;
+
+/**
+ * @author Dmitrii Tikhomirov <chani@me.com> Created by treblereel on 10/26/18.
+ */
+public class AptContext {
+
+    public final Messager messager;
+    public final Filer filer;
+    public final Elements elements;
+    public final Types types;
+    public final ProcessingEnvironment processingEnv;
+
+    public AptContext(final ProcessingEnvironment processingEnv) {
+        this.filer = processingEnv.getFiler();
+        this.messager = processingEnv.getMessager();
+        this.elements = processingEnv.getElementUtils();
+        this.types = processingEnv.getTypeUtils();
+        this.processingEnv = processingEnv;
+    }
+
+    /**
+     * A wrapper over {@link Types#isAssignable(TypeMirror, TypeMirror)} which will apply type erasure on the targetClass before calling the wrapped method..
+     *
+     * @param element  a {@link javax.lang.model.element.Element} object.
+     * @param targetClass a {@link java.lang.Class} object.
+     * @return a boolean.
+     */
+    public boolean isAssignableFrom(Element element, Class<?> targetClass) {
+        return isAssignableFrom(element.asType(), targetClass);
+    }
+
+    /**
+     * A wrapper over {@link Types#isAssignable(TypeMirror, TypeMirror)} which will apply type erasure on the targetClass before calling the wrapped method.
+     *
+     * @param typeMirror  a {@link javax.lang.model.type.TypeMirror} object.
+     * @param targetClass a {@link java.lang.Class} object.
+     * @return a boolean.
+     */
+    public boolean isAssignableFrom(TypeMirror typeMirror, Class<?> targetClass) {
```

Comment:
Methods like this one could be moved to something like TypeUtils (domino-jackson)

Reply:
I think this already part of domino-jackson

Reply:
We dont share this with other modules